crowdfunding/projects/views.py

```diff
@@ -1,3 +1,143 @@
 from django.shortcuts import render
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from django.http import Http404, JsonResponse
+from rest_framework import status, generics, permissions, filters
+
+from .models import Project, Pledge
+from .serializers import ProjectSerializer, PledgeSerializer, ProjectDetailSerializer, PledgeDetailSerializer
+
+from .permissions import IsOwnerOrReadOnly, IsSupporterOrReadOnly
+
 
 # Create your views here.
+
+class ProjectList(generics.ListCreateAPIView):
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly, IsOwnerOrReadOnly]
+    queryset = Project.objects.all()
+    serializer_class = ProjectSerializer
+    filter_backends = [filters.SearchFilter]
+
+    search_fields = ['title', 'description']
+
+    def perform_create(self, serializer):
+        serializer.save(owner=self.request.user)
+
+    def perform_update(self, serializer):
+        serializer.save(owner=self.request.user)
```

Comment:
Absolutely LOVE this, great example of you understanding how the generic classes work.